gobblin-utility/src/main/java/gobblin/util/ProxiedFileSystemWrapper.java

```diff
@@ -0,0 +1,151 @@
+    String proxyUserName = properties.getProp(ConfigurationKeys.FS_PROXY_AS_USER_NAME);
+    UserGroupInformation proxyUser;
+    switch (authType) {
+      case KEYTAB: // If the authentication type is KEYTAB, log in a super user first before creating a proxy user.
+        Preconditions.checkArgument(
+            StringUtils.isNotBlank(properties.getProp(ConfigurationKeys.SUPER_USER_NAME_TO_PROXY_AS_OTHERS)),
+            "State does not contain a proper proxy token file name");
+        String superUser = properties.getProp(ConfigurationKeys.SUPER_USER_NAME_TO_PROXY_AS_OTHERS);
+        UserGroupInformation.loginUserFromKeytab(superUser, authPath);
+        proxyUser = UserGroupInformation.createProxyUser(proxyUserName, UserGroupInformation.getLoginUser());
+        break;
+      case TOKEN: // If the authentication type is TOKEN, create a proxy user and then add the token to the user.
+        proxyUser = UserGroupInformation.createProxyUser(proxyUserName, UserGroupInformation.getLoginUser());
+        Optional<Token> proxyToken = this.getTokenFromSeqFile(authPath, proxyUserName);
+        if (proxyToken.isPresent()) {
+          proxyUser.addToken(proxyToken.get());
+        } else {
+          LOG.warn("No delegation token found for the current proxy user.");
+        }
+        break;
+      default:
+        LOG.warn("Creating a proxy user without authentication, which could not perform File system operations.");
+        proxyUser = UserGroupInformation.createProxyUser(proxyUserName, UserGroupInformation.getLoginUser());
+        break;
+    }
+
+    final Configuration conf = new Configuration();
+    JobConfigurationUtils.putStateIntoConfiguration(properties, conf);
+    final URI fsURI = URI.create(uri);
+    proxyUser.doAs(new PrivilegedExceptionAction<Void>() {
+      @Override
+      public Void run() throws IOException {
+        LOG.info("Now performing file system operations as :" + UserGroupInformation.getCurrentUser());
+        proxiedFs = FileSystem.get(fsURI, conf);
+        return null;
+      }
+    });
+    return this.proxiedFs;
+  }
+
+  /**
+   * Get token from the token sequence file.
+   * @param authPath
+   * @param proxyUserName
+   * @return Token for proxyUserName if it exists.
+   */
+  private Optional<Token> getTokenFromSeqFile(String authPath, String proxyUserName) {
+    Closer closer = Closer.create();
+    try {
+      FileSystem localFs = FileSystem.get(new URI(ConfigurationKeys.LOCAL_FS_URI), new Configuration());
+      SequenceFile.Reader tokenReader =
+          closer.register(new SequenceFile.Reader(localFs, new Path(authPath), localFs.getConf()));
+      Text key = new Text();
+      Token value = new Token();
+      while (tokenReader.next(key, value)) {
+        LOG.info("Find token for " + key);
+        if (key.toString().equals(proxyUserName)) {
+          return Optional.of(value);
+        }
+      }
+
+    } catch (IOException e) {
+      Throwables.propagate(e);
```

Comment:
Ditto. I suggest allowing the method itself to throw at least `IOException` instead of using `propagate`.